src/main/java/com/coveros/selenified/services/Assert.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2019 Coveros, Inc.
+ *
+ * This file is part of Selenified.
+ *
+ * Selenified is licensed under the Apache License, Version
+ * 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy
+ * of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.coveros.selenified.services;
+
+import com.coveros.selenified.utilities.Reporter;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import java.util.Map;
+
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.assertTrue;
+
+/**
+ * Assert will handle all verifications performed on the actual web services
+ * calls themselves. These asserts are custom to the framework, and in addition to
+ * providing easy object oriented capabilities, they assist in
+ * troubleshooting and debugging failing tests.
+ *
+ * @author Max Saperstone
+ * @version 3.2.0
+ * @lastupdate 4/4/2019
+ */
+public class Assert implements Check {
+
+
+    // this will be the name of the file we write all commands out to
+    private final Reporter reporter;
+
+    // this is the driver that will be used for all selenium actions
+    private final Response response;
+
+    /**
+     * The default constructor passing in the app and output file
+     *
+     * @param response - the response from the web services call
+     * @param reporter - the file to write all logging out to
+     */
+    public Assert(Response response, Reporter reporter) {
+        this.response = response;
+        this.reporter = reporter;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Reporter getReporter() {
+        return reporter;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Response getResponse() {
+        return response;
+    }
+
+    ///////////////////////////////////////////////////////
+    // assertions about the page in general
+    ///////////////////////////////////////////////////////
+
+    /**
+     * Asserts the actual response code is equals to the expected response
+     * code, and writes that out to the output file. If this fails, the code will
+     * immediately exit, and record the error.
+     *
+     * @param expectedCode - the expected response code
+     */
+    @Override
+    public void equals(int expectedCode) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either override Object.equals(Object), or rename the method to prevent any confusion. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS1201)